rand_distr/src/normal.rs

```diff
@@ -186,22 +224,78 @@ where F: Float, StandardNormal: Distribution<F>
 where F: Float, StandardNormal: Distribution<F>
 {
-    /// Construct a new `LogNormal` distribution with the given mean
-    /// and standard deviation of the logarithm of the distribution.
+    /// Construct, from (log-space) mean and standard deviation
+    ///
+    /// Parameters are the "standard" log-space measures (these are the mean
+    /// and standard deviation of the logarithm of samples):
+    ///
+    /// -   `mu` (`μ`, unrestricted) is the mean of the underlying distribution
+    /// -   `sigma` (`σ`, must be finite) is the standard deviation of the
+    ///     underlying Normal distribution
+    #[inline]
+    pub fn new(mu: F, sigma: F) -> Result<LogNormal<F>, Error> {
+        let norm = Normal::new(mu, sigma)?;
+        Ok(LogNormal { norm })
+    }
+
+    /// Construct, from (linear-space) mean and coefficient of variation
+    ///
+    /// Parameters are linear-space measures:
+    ///
+    /// -   mean (`μ > 0`) is the (real) mean of the distribution
+    /// -   coefficient of variation (`cv = σ / μ`, requiring `cv ≥ 0`) is a
+    ///     standardized measure of dispersion
+    ///
+    /// As a special exception, `μ = 0, cv = 0` is allowed (samples are `-inf`).
     #[inline]
-    pub fn new(mean: F, std_dev: F) -> Result<LogNormal<F>, Error> {
-        if !(std_dev >= F::zero()) {
-            return Err(Error::StdDevTooSmall);
+    pub fn from_mean_cv(mean: F, cv: F) -> Result<LogNormal<F>, Error> {
+        if cv == F::zero() {
+            let mu = mean.ln();
+            let norm = Normal::new(mu, F::zero()).unwrap();
+            return Ok(LogNormal { norm });
         }
-        Ok(LogNormal {
-            norm: Normal::new(mean, std_dev).unwrap(),
-        })
+        if !(mean > F::zero()) {
+            return Err(Error::MeanTooSmall);
+        }
+        if !(cv >= F::zero()) {
+            return Err(Error::BadVariance);
+        }
+
+        // Using X ~ lognormal(μ, σ), CV² = Var(X) / E(X)²
+        // E(X) = exp(μ + σ² / 2) = exp(μ) × exp(σ² / 2)
+        // Var(X) = exp(2μ + σ²)(exp(σ²) - 1) = E(X)² × (exp(σ²) - 1)
+        // but Var(X) = (CV × E(X))² so CV² = exp(σ²) - 1
+        // thus σ² = log(CV² + 1)
+        // and exp(μ) = E(X) / exp(σ² / 2) = E(X) / sqrt(CV² + 1)
+        let a = F::one() + cv * cv; // e
+        // let mu = F::from(0.5).unwrap() * (mean * mean / a).ln();
+        let mu = (mean / a.sqrt()).ln();
+        let sigma = a.ln().sqrt();
+        let norm = Normal::new(mu, sigma)?;
+        Ok(LogNormal { norm })
+    }
+
+    /// Sample from a z-score
+    ///
+    /// This may be useful for generating correlated samples, as follows.
+    /// ```
+    /// # use rand::prelude::*;
+    /// # use rand_distr::{LogNormal, StandardNormal};
+    /// let mut rng = thread_rng();
+    /// let z = StandardNormal.sample(&mut rng);
+    /// let x1 = LogNormal::from_mean_cv(3.0, 1.0).unwrap().from_zscore(z);
+    /// let x2 = LogNormal::from_mean_cv(2.0, 1.0).unwrap().from_zscore(z);
+    /// ```
```

Comment:
```suggestion
    /// This may be useful for generating correlated samples `x1` and `x2`
    /// from two different distributions, as follows.
    /// ```
    /// # use rand::prelude::*;
    /// # use rand_distr::{LogNormal, StandardNormal};
    /// let mut rng = thread_rng();
    /// let z = StandardNormal.sample(&mut rng);
    /// let x1 = LogNormal::from_mean_cv(3.0, 1.0).unwrap().from_zscore(z);
    /// let x2 = LogNormal::from_mean_cv(2.0, 4.0).unwrap().from_zscore(z);
    /// ```
```